packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -390,19 +390,63 @@ class FlutterPlugin implements Plugin<Project> {
      * Configures the Flutter plugin dependencies.
      *
      * The plugins are added to pubspec.yaml. Then, upon running `flutter pub get`,
-     * the tool generates a `.flutter-plugins` file, which contains a 1:1 map to each plugin location.
+     * the tool generates a `.flutter-plugins-dependencies` file, which contains a map to each plugin location.
      * Finally, the project's `settings.gradle` loads each plugin's android directory as a subproject.
      */
     private void configurePlugins() {
+        configureLegacyPluginProjects()
         getPluginList().each this.&configurePluginProject
-        getPluginDependencies().each this.&configurePluginDependencies
+        getPluginList().each this.&configurePluginDependencies
+    }
+
+    // TODO(gustl22): Can remove configuring legacy plugins, once #54566 and therefor #48918 is resolved.
+    //  This means all references to `.flutter-plugins` are then removed and apps only depend exclusively on the
+    //  `plugins` property in `.flutter-plugins-dependencies`.
+    //  This cannot be achieved yet as developers may still use an old `settings.gradle`
+    //  which includes all the plugins from the `.flutter-plugins` file, even if not made for Android. 
+    //  The settings.gradle then:
+    //      1) tries to add the android plugin implementation, which does not exist at all, but is also not included
+    //         successfully (which does not throw an error and therefore isn't a problem), or
+    //      2) includes the plugin successfully as a valid android plugin directory exists, even if the surrounding
+    //         flutter package does not support the android platform (see e.g. apple_maps_flutter: 1.0.1). 
+    //         So as it's included successfully it expects to be added as API. This is only possible by taking all 
+    //         plugins into account, which only appear on the `dependencyGraph` and under the `plugins`.
+    //  So in summary the plugins are currently selected from the `dependencyGraph` and filtered then with the
+    //  [doesSupportAndroidPlatform] method instead of just using the `plugins.android` list.
+    private configureLegacyPluginProjects() {
+        File settingsGradle = new File(project.projectDir.parentFile, 'settings.gradle')
+        if(!settingsGradle.exists() || !settingsGradle.text.contains("'.flutter-plugins'")) {
+            return
+        }
+        project.logger.error("Your app uses an outdated version of `settings.gradle`. Please update.")
```

Comment:
This error message should link to something the developer can use to understand how to update, what timeline they have to update and what happens if they dont.

Reply:
Moved via b7901cd7a9f9ca88364202aa6c89461eaecc1049